thirdparties/glog/glog.patch

```diff
@@ -741,7 +742,30 @@ index 0000000..1f3c8c1
 +    }
 +}
 +
++// NOTE: this function only run once in child process
++void AsyncLogger::ResetInChildProcess(bool start) {
++    if (!PidHasChanged()) {
++        return;
++    }
++
++    // After fork(2) and in the child process we should:
++    //   (1) reset all status for AyncLogger
++    //   (2) restart AyncLogger when necessary
++    appThreadsBlockedCount_ = 0;
++    flushCount_ = 0;
++    mutex_.unlock();
++    activeBuffer_.reset(new Buffer());
++    flushBuffer_.reset(new Buffer());
++    running_ = false;
++
++    if (start) {
++        Start();
++    }
++}
++
 +void AsyncLogger::Start() {
++    ResetInChildProcess(false);
```

Comment:
AsyncLogger::Start() will called by child process after fork?

Reply:
> 

yep, because in the child process, there is no SyncLogger thread.